src/fileseq/filesequence.py

```diff
@@ -718,124 +914,200 @@ def findSequenceOnDisk(cls, pattern, strictPadding=False):
 
         Yields:
             str:
         """
-        _check = cls.DISK_RE.match
+        if decimal_places == 0:
+            _check = cls.DISK_RE.match
+        else:
+            _check = cls.DISK_SUB_RE.match
 
         for item in iterable:
             # Add a filter for paths that don't match the frame
             # padding of a given number
             matches = _check(item)
             if not matches:
-                if num <= 0:
+                if zfill is None or zfill <= 0:
                     # Not a sequence pattern, but we were asked
                     # to match on a zero padding
                     yield item
 
                 continue
 
+            # Ensure DISK_RE matches before calling optional get_frame function
             frame = matches.group(3) or ''
+            if frame and get_frame is not None:
+                frame = get_frame(item) or ''
 
             if not frame:
-                if num <= 0:
+                if zfill is None or zfill <= 0:
                     # No frame value was parsed, but we were asked
                     # to match on a zero padding
                     yield item
                 continue
 
             # We have a frame number
+            frame, _, subframe = frame.partition(".")
+            if len(subframe) != decimal_places:
+                continue
+
+            if zfill is None:
+                yield item
+                continue
 
             if frame[0] == '0' or frame[:2] == '-0':
-                if len(frame) == num:
+                if len(frame) == zfill:
                     # A frame leading with '0' is explicitly
                     # padded and can only be a match if its exactly
                     # the target padding number
                     yield item
                 continue
 
-            if len(frame) >= num:
+            if len(frame) >= zfill:
                 # A frame that does not lead with '0' can match
                 # a padding width >= to the target padding number
                 yield item
                 continue
 
     @staticmethod
-    def getPaddingChars(num):
+    def getPaddingChars(num, pad_style=PAD_STYLE_DEFAULT):
         """
         Given a particular amount of padding, return the proper padding characters.
 
         Args:
-            num (int):
+            num (int): required width of string with padding
+            pad_style (PAD_STYLE_DEFAULT or PAD_STYLE_HASH1 or PAD_STYLE_HASH4): padding style
 
         Returns:
             str:
         """
-        if num == 0:
-            return "@"
-        if num % 4 == 0:
-            return "#" * (num // 4)
-        else:
-            return "@" * num
+        if pad_style is PAD_STYLE_HASH1:
+            return "#" * max(1, num)
+        elif pad_style is PAD_STYLE_HASH4:
+            if num == 0:
+                return "@"
+            if num % 4 == 0:
+                return "#" * (num // 4)
+            else:
+                return "@" * num
```

Comment:
Seems like this kind of duplicates the definitions of the constants and pad map. Would it make more sense for this function to not know that PAD_STYLE_HASH1 is 1 and PAD_STYLE_HASH4 and to just look them up from constants? In gofileseq it defers to a pad mapper implementation for each type to do all the conversions.

Reply:
Yeah, that is nicer in pad.cpp.

I figured that the existing code already had knowledge of the pad characters split this way (some knowledge in constants.py and some in FileSequence.getPaddingChars), so I wasn't making that any worse, and by this point I was getting a bit lazy. The code in constants.py goes to a bit of effort to construct regular expressions using PAD_MAP, so I'm not sure of the appropriate place to put pad mapping classes to keep the knowledge together where some is code rather than pure constants.

Note that even gofileseq cheats a little, constructing regular expressions with hard-coded knowledge of the padding characters :)

Reply:
Yea fair enough. Not worth holding up a merge. I can get this merged in a release a 1.11.0

Reply:
Haha, to be clear, totally happy to make it nicer, just wondering if you have any suggestions? constants.py doesn't seem like the right place to put a function.

Maybe a dictionary mapping style to default character etc, equivalent to the pad mapping subclass content?

Reply:
Ah, I just figured we wouldn't mess with it. I already tagged 1.11.0 but I havent pushed it to pypi yet. Let me know if you want to push a dictionary mapping and I can hold off.

Reply:
It's an implementation details that doesn't affect the API, so not worth holding up a release. I'll have a think about it and make a separate PR, unless you beat me to it.